test/test-BlueprintParser.cc

```diff
@@ -779,11 +779,11 @@ TEST_CASE("Report error when there are circular references in nested members", "
     ParseResult<Blueprint> blueprint;
     SectionParserHelper<Blueprint, BlueprintParser>::parse(source, BlueprintSectionType, blueprint, ExportSourcemapOption, Models(), &blueprint);
 
-    REQUIRE(blueprint.report.error.code == MSONError);
-    SourceMapHelper::check(blueprint.report.error.location, 50, 7);
+    REQUIRE(blueprint.report.error.code == Error::OK);
+    REQUIRE(blueprint.report.warnings.empty());
```

Comment:
Here we need to add `REQUIRE`s to check that base type of `A`, `B` and `C` are `mson::ImplicitObjectBaseType`.